src/library-authoring/component-info/ComponentPreview.tsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import { useIntl } from '@edx/frontend-platform/i18n';
+import { Button, StandardModal, useToggle } from '@openedx/paragon';
+import { OpenInFull } from '@openedx/paragon/icons';
+
+import { LibraryBlock } from '../LibraryBlock';
+import messages from './messages';
+
+// This is a simple overlay to prevent interaction with the preview
+const PreviewOverlay = () => (
+  <div className="position-absolute w-100 h-100 zindex-9" />
+);
+
+interface ModalComponentPreviewProps {
+  isOpen: boolean;
+  close: () => void;
+  usageKey: string;
+}
+
+const ModalComponentPreview = ({ isOpen, close, usageKey }: ModalComponentPreviewProps) => {
+  const intl = useIntl();
+
+  return (
+    <StandardModal
+      title={intl.formatMessage(messages.previewModalTitle)}
+      isOpen={isOpen}
+      onClose={close}
+      footerNode={(
+        <Button variant="tertiary" onClick={close}>
+          {intl.formatMessage(messages.previewModalCloseButtonTitle)}
+        </Button>
+      )}
+      className="component-preview-modal"
+    >
+      <LibraryBlock usageKey={usageKey} />
+    </StandardModal>
+  );
+};
+
+interface ComponentPreviewProps {
+  usageKey: string;
+}
+
+const ComponentPreview = ({ usageKey }: ComponentPreviewProps) => {
+  const intl = useIntl();
+
+  const [isModalOpen, openModal, closeModal] = useToggle();
+
+  return (
+    <>
+      <div className="position-relative m-2">
+        <PreviewOverlay />
+        <Button
+          size="sm"
+          variant="light"
+          iconBefore={OpenInFull}
+          onClick={openModal}
+          className="position-absolute right-0 zindex-10 m-1"
+        >
+          {intl.formatMessage(messages.previewExpandButtonTitle)}
+        </Button>
+        <LibraryBlock usageKey={usageKey} />
```

Comment:
In the future, it would be awesome to use something like [react-reverse-portal](https://github.com/httptoolkit/react-reverse-portal) to "re-parent" the `<LibraryBlock>` so that it _moves_ from the sidebar to the modal instead of being _copied_ into the modal. Then for example, you could start playing a video in the sidebar and "Expand" it and it would still be playing in the modal. Please mention that to the UX team as a possible future improvement after this merges (and post MVP).

Reply:
Sure! I will mention that!